test/compress/conditionals.js

```diff
@@ -344,7 +344,7 @@ cond_8: {
 
         a = condition ? !0 : !1;
         a = !condition ? !null : !2;
-        a = condition() ? !0 : !-3.5;
+        a = condition() ? !0 : !3.5;
```

Comment:
I don't believe this change was intentional, so I'll hold of merging to let @kzc verify.